artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/proton/AMQPLargeMessageWriter.java

```diff
@@ -81,11 +84,20 @@ public boolean isWriting() {
    public void close() {
       if (!closed) {
          try {
+            try {
+               if (largeBodyReader != null) {
+                  largeBodyReader.close();
+               }
+               largeBodyReader = null;
```

Comment:
Should be nulled in a finally block, which would happen in the resetClosed method called in the main finally block if you added a null there for the large body reader as you should.